lib/backend/lite/lite.go

```diff
@@ -0,0 +1,845 @@
+// configuration struct as a parameter
+func NewWithConfig(ctx context.Context, cfg Config) (*LiteBackend, error) {
+	if err := cfg.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// Ensure that the path to the root directory exists.
+	err := os.MkdirAll(cfg.Path, defaultDirMode)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	fullPath := filepath.Join(cfg.Path, defaultDBFile)
+	connectorURL := fmt.Sprintf("file:%v?_busy_timeout=%v&_sync=%v", fullPath, cfg.BusyTimeout, cfg.Sync)
+	db, err := sql.Open(BackendName, connectorURL)
+	if err != nil {
+		return nil, trace.Wrap(err, "error opening URI: %v", fullPath)
+	}
+	// serialize access to sqlite to avoid database is locked errors
+	db.SetMaxOpenConns(1)
+	buf, err := backend.NewCircularBuffer(ctx, cfg.BufferSize)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	closeCtx, cancel := context.WithCancel(ctx)
+	watchStarted, signalWatchStart := context.WithCancel(ctx)
+	l := &LiteBackend{
+		Config:           cfg,
+		db:               db,
+		Entry:            log.WithFields(log.Fields{trace.Component: BackendName}),
+		clock:            cfg.Clock,
+		buf:              buf,
+		ctx:              closeCtx,
+		cancel:           cancel,
+		watchStarted:     watchStarted,
+		signalWatchStart: signalWatchStart,
+	}
+	l.Debugf("Connected to: %v", connectorURL)
+	if err := l.createSchema(); err != nil {
+		return nil, trace.Wrap(err, "error creating schema: %v", fullPath)
+	}
+	if err := l.showPragmas(closeCtx); err != nil {
+		l.Warningf("Failed to show pragma settings: %v.", err)
+	}
+	go l.runPeriodicOperations()
+	return l, nil
+}
+
+// LiteBackend uses SQLite to implement storage interfaces
+type LiteBackend struct {
+	Config
+	*log.Entry
+	db *sql.DB
+	// tx stores the current transaction if this backend instance
+	// is bound to transaction
+	tx *sql.Tx
+	// clock is used to generate time,
+	// could be swapped in tests for fixed time
+	clock clockwork.Clock
+
+	buf              *backend.CircularBuffer
+	ctx              context.Context
+	cancel           context.CancelFunc
+	watchStarted     context.Context
+	signalWatchStart context.CancelFunc
+}
+
+// showPragmas is used to debug SQLite database connection
+// parameters, when called, logs some key PRAGMA values
+func (l *LiteBackend) showPragmas(ctx context.Context) error {
+	return l.inTransaction(ctx, func(tx *sql.Tx) error {
+		row := tx.QueryRowContext(ctx, "PRAGMA synchronous;")
+		var value string
+		if err := row.Scan(&value); err != nil {
+			return trace.Wrap(err)
+		}
+		l.Debugf("Synchronous: %v", value)
+
+		row = tx.QueryRowContext(ctx, "PRAGMA busy_timeout;")
+		if err := row.Scan(&value); err != nil {
+			return trace.Wrap(err)
+		}
+		l.Debugf("Busy timeout: %v", value)
+		return nil
+	})
+}
+
+func (l *LiteBackend) createSchema() error {
+	schemas := []string{
+
+		`CREATE TABLE IF NOT EXISTS kv (
+           key TEXT NOT NULL PRIMARY KEY,
```

Comment:
I think `byte` is a better  container for this key, as there is nothing specifically `string` like about it - it can contain random combinations of non-printable characters or non-unicode characters for example.